### PR TITLE
feat: Add System Resource instead of $resources url

### PR DESCRIPTION
**Pull request by istarkov**
## Description

ref #3308

image

image

# TODO:

- [x] - Remove copy for curl
image

## Steps for reproduction

Add Variable, System Resource, open Select, check texts

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: packages/sdk/src/schema/resources.ts [hunk 1]
```diff
@@ -49,6 +49,16 @@ const LOCAL_RESOURCE_PREFIX = "$resources";
...
+  const segments = pathname.split("/").filter(Boolean);
+
+  if (resourceName === undefined) {
+    return segments[0] === LOCAL_RESOURCE_PREFIX;
+  }
+
+  return segments.join("/") === `${LOCAL_RESOURCE_PREFIX}/${resourceName}`;
+};
+
+export const sitemapResourceUrl = () => {
```
Comment (TrySound) [inline packages/sdk/src/schema/resources.ts:62]: could be just value, not function

File: apps/builder/app/builder/features/settings-panel/resource-panel.tsx [hunk 2]
```diff
@@ -630,3 +626,96 @@ export const ResourceForm = forwardRef<
...
+      );
+    },
+  }));
+
+  const resourceId = useId();
+
+  return (
+    <>
+      <Flex direction="column" css={{ gap: theme.spacing[3] }}>
+        <Label htmlFor={resourceId}>Resource</Label>
```
Comment (TrySound) [inline apps/builder/app/builder/features/settings-panel/resource-panel.tsx:702]: doesn't work anyway, resourceId can be removed

**Review by TrySound — APPROVED** [review]

File: apps/builder/app/builder/features/settings-panel/variable-popover.tsx [hunk 3]
```diff
@@ -427,7 +437,20 @@ const VariablePanel = forwardRef<
...
+    [
+      "system-resource",
+      {
+        label: (
+          <Flex direction="row" gap="2" align="center">
+            System Resource
+            {allowDynamicData === false && <ProBadge>Pro</ProBadge>}
+          </Flex>
+        ),
+        description: "A System Resource is a configuration for system data.",
```
Comment (kof) [inline apps/builder/app/builder/features/settings-panel/variable-popover.tsx:449]: not sure if much better, but seems like more clear which system it is

```suggestion
        description: "A System Resource is a configuration for Webstudio data.",
```

**Review by kof — APPROVED** [review]
🎉

File: apps/builder/app/builder/features/settings-panel/resource-panel.tsx (hunk 2 above)
Comment (istarkov) [reply apps/builder/app/builder/features/settings-panel/resource-panel.tsx:702]: May be will in the future

**Pull request closed without merging**